public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -408,6 +410,10 @@ export function buildGridItemForPanel(panel: PanelModel): DashboardGridItem {
     $behaviors: [],
     extendPanelContext: setDashboardPanelContext,
     _UNSAFE_customMigrationHandler: getAngularPanelMigrationHandler(panel),
+    headerActions:
+      panel.fieldConfig?.defaults?.timeCompare && isTimeCompareSupported(panel.type)
+        ? [new CustomTimeRangeCompare({ key: 'time-compare', compareWith: undefined, compareOptions: [] })]
+        : undefined,
```

Comment:
I supopse this is mainly about checking if panel has `timeCompare === true`. Not sure why we need a registry of panels that support the comparison. If we want to enable it for time series panel fir instance, let's just define `timeCompare` option in the plugin definition, similarly to how we add any other option (https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/timeseries/panelcfg.gen.ts#L13)